force-app/main/default/classes/CountTasksTest.cls

```diff
@@ -0,0 +1,188 @@
+/**
+ * Apex Test Class for all scenarios
+ * (insert, update, delete, close task,
+ * re open task, reparenting task)
+ */
+@isTest
+public class CountTasksTest {
+  private static final Integer ACCOUNTS_NUMBER = 180;
+  private static final Integer TASKS_NUMBER = 13;
+
+  @TestSetup
+  private static void makeData() {
+    List<Account> accounts = new List<Account>();
+
+    for (Integer i = 0; i < ACCOUNTS_NUMBER; i++) {
+      Account acc = new Account(
+        Name = 'Test Account ' + i,
+        Type = 'Prospect',
+        RecordTypeId = TaskTriggerHandler.recordTypeUS
+      );
+      accounts.add(acc);
+    }
+
+    insert accounts;
+
+    List<Task> tasks = new List<Task>();
+    for (Account acc : accounts) {
+      for (Integer i = 0; i < TASKS_NUMBER; i++) {
+        Task tskCall = new Task(
+          Subject = 'Call',
+          Type = 'Call',
+          WhatId = acc.Id
+        );
+        tasks.add(tskCall);
+        Task tskEmail = new Task(
+          Subject = 'Email',
+          Type = 'Email',
+          WhatId = acc.Id
+        );
+        tasks.add(tskEmail);
+        Task tskMeeting = new Task(
+          Subject = 'Meeting',
+          Type = 'Meeting',
+          WhatId = acc.Id
+        );
+        tasks.add(tskMeeting);
+        Task tskOther = new Task(
+          Subject = 'Other',
+          Type = 'Other',
+          WhatId = acc.Id
+        );
+        tasks.add(tskOther);
+      }
+    }
+    insert tasks;
+  }
+
+  @isTest
+  private static void shouldHaveAccountsAndTasksInDatabase() {
+    Assert.areEqual(
+      ACCOUNTS_NUMBER,
+      [SELECT COUNT() FROM Account],
+      'Failed to get expected amount of Accounts'
+    );
+    Integer numberOfTaskTypes = 4;
+    Assert.areEqual(
+      (ACCOUNTS_NUMBER * TASKS_NUMBER * numberOfTaskTypes),
+      [SELECT COUNT() FROM Task],
+      'Failed to get expected amount of tasks'
+    );
+  }
+
+  //insert scenario
+  @isTest
+  public static void insertNewTaskAccount() {
```

Comment:
Usually, we strive to bulkify our code.

So this test should ideally create several tasks for several accounts at once

Reply:
Ok...but do I really need this method? Am I not already testing the 'insert' case with the makeData() method-the method with @TestSetup annotation?

Reply:
I've updated the test class. I know you said it doesn't matter for now if I do it, but it's something to keep in mind for the future, but I wanted to try to figure out if I’m on the right track. 

As for the flows: in some places I used AccountId and not WhatId (to select accounts related to Tasks), so that's something I need to change because we said that AccountId cann't be used for our purpose.

Currently the flow is triggered by tasks that belong to any account and this can easily exceed governor limits.  So that's certainly something that need to be changed, because we need to select only the accounts with certain characteristics.
Finally, I think the business logic is a bit complex and the best solution in this case is the Trigger and not the Record-triggered flow.

Reply:
For the future triggers that I will create, I will take into consideration all your suggestions (division between Trigger, Helper, Handler and Service classes). Thank you!